CsmakeModules/SystemBuildMsdosPartitions.py

```diff
@@ -91,12 +102,19 @@ class SystemBuildMsdosPartitions(CsmakeModule):
         'linux' : 'ext2',
         'fat16' : 'fat16',
         'fat32' : 'fat32',
+        'vfat' : 'fat32',
         'HFS' : 'HFS',
         'NTFS' : 'NTFS',
         'linux-swap' : 'linux-swap',
         'reiserfs' : 'reiserfs',
         'ufs' : 'ufs' }
 
+    PART_TYPE = "msdos"
+    PART_PRIMARY_PARTITIONS = 4
+    PART_LOGICAL_EXTENDED_ALLOWED = True
+    #Tell parted to start on sector 2048
+    PART_FIRST_START = "2048s"
```

Comment:
This is a great addition that allows us to actually start at "0" as far as parted is concerned - this is a workaround to get around the problem where parted wants to carve out the first 1024 as a special partition.
This really oughtn't be necessary - especially for msdos partitioning schemes, but some versions of parted are hardwired to operate at 2048 and beyond - so this works more generally, and doesn't chew up 1% of the beginning of the disk like the old scheme did.